source/ui/CLS/CLSSIS3820ScalerDarkCurrentWidget.cpp

```diff
@@ -74,23 +77,43 @@ void CLSSIS3820ScalerDarkCurrentWidget::setScaler(CLSSIS3820Scaler *newScaler)
 		scaler_ = newScaler;
 
 		if (scaler_) {
-			connect( scaler_, SIGNAL(scanningChanged(bool)), this, SLOT(onScalerScanningChanged()) );
-
-			timeEntry_->setValue((int)scaler_->dwellTime() * MILLISECONDS_PER_SECOND);
+			connect( scaler_, SIGNAL(connectedChanged(bool)), this, SLOT(refresh()) );
+			connect( scaler_, SIGNAL(dwellTimeChanged(double)), this, SLOT(updateTimeBox()) );
+			connect( scaler_, SIGNAL(scanningChanged(bool)), this, SLOT(updateCollectButton()) );
 		}
 
-		onScalerScanningChanged();
+		refresh();
 
 		emit scalerChanged(scaler_);
 	}
 }
 
-void CLSSIS3820ScalerDarkCurrentWidget::onScalerScanningChanged()
+void CLSSIS3820ScalerDarkCurrentWidget::updateTimeBox()
 {
-	if (scaler_ && !scaler_->isScanning()) {
+	if (scaler_ && scaler_->isConnected()) {
+
+		timeBox_->blockSignals(true);
+		timeBox_->setEnabled(true);
+		timeBox_->setRange(0, INT_MAX);
+		timeBox_->setSuffix(" ms");
+		timeBox_->setValue(int(scaler_->dwellTime() * 1000));
```

Comment:
I'd say so.  Everything in AM uses seconds as its base.  Even the scaler model.  We should probably endeavour to get the scaler view in line as well.